src/components/clusterprovider/minikube/minikube.ts

```diff
@@ -83,31 +89,44 @@ class MinikubeImpl implements Minikube {
     }
 }
 
-async function minikubeUpgradeAvailable(context: Context): Promise<void> {
-
-    const performUpgradeCheck = await checkPresent(context, CheckPresentMode.Silent) && getCheckForMinikubeUpgrade();
-    if (!performUpgradeCheck) {
-        return;
-    }
-
+async function getVersionInfo(context: Context): Promise<MinikubeVersionInfo> {
     const sr = await context.shell.exec(`"${context.binPath}" update-check`);
     if (!sr || sr.code !== 0) {
-        vscode.window.showErrorMessage(`Error checking for minikube updates: ${sr ? sr.stderr : 'cannot run minikube'}`);
-        return;
+        throw new Error(`Error checking for minikube updates: ${sr ? sr.stderr : 'cannot run minikube'}`);
     }
     const lines = sr.stdout.split('\n')
                            .map((l) => l.trim())
                            .filter((l) => l.length > 0);
     if (lines.length !== 2) {
-        vscode.window.showErrorMessage(`Unexpected output for minikube version check: ${lines}`);
+        throw new Error(`Unexpected output for minikube version check: ${lines}`);
+    }
+
+    const currentVersion = semver.clean(extractVersion(lines[0]));
+    const availableVersion = semver.clean(extractVersion(lines[0]));
+
+    if (currentVersion === null || availableVersion === null) {
+        throw new Error(`Unable to get version from minikube version check: ${lines}`);
+    }
+
+    return {
+        currentVersion: currentVersion,
+        availableVersion: availableVersion
+    } as MinikubeVersionInfo;
+}
+
+async function minikubeUpgradeAvailable(context: Context): Promise<void> {
+
+    const performUpgradeCheck = await checkPresent(context, CheckPresentMode.Silent) && getCheckForMinikubeUpgrade();
+    if (!performUpgradeCheck) {
         return;
     }
-    const currentVersion = extractVersion(lines[0]);
-    const availableVersion = extractVersion(lines[1]);
-    if (currentVersion !== availableVersion) {
-        const value = await vscode.window.showInformationMessage(`Minikube upgrade available to ${availableVersion}, currently on ${currentVersion}`, 'Install');
+
+    const versionInfo = await getVersionInfo(context);
```

Comment:
`getVersionInfo` has some explicit `throw`s in it - should this be in a try block?

Reply:
i think https://github.com/Azure/vscode-kubernetes-tools/pull/575/commits/504284da3bbc78122cfb00ce429fd09f9679a701 will do